packages/ai-bot/main.ts

```diff
@@ -326,12 +329,14 @@ Common issues are:
 
   //handle set title by commands
   client.on(RoomEvent.Timeline, async function (event, room) {
-    if (!room) {
-      return;
-    }
-    if (!isCommandResultStatusApplied(event)) {
+    if (
+      event.event.origin_server_ts! < startTime ||
```

Comment:
Don't we want to try to process events that might have come in while the container is rebooting (i.e. during deploys)? @IanCal @jurgenwerk

Reply:
Perhaps some grace time would be useful, but on the other hand, would it not be a bit odd to receive answers to your prompts/commands in a delayed manner? Depends on how long the deploy/switchover process takes, if it's only a couple of seconds then it would make sense.

I think this is part of a larger question of how to handle graceful shutdowns and rolling deploys of the AI bot. If we do that properly then we won't have to think at what time the events are arriving.

Something like:
- when a new version is deployed, the load balancer should direct the traffic to the new instance
- in the "old" instance, wait for the AI response (streaming) to be fully done before terminating it